File: backend/app/schemas/pricing.py
from pydantic import BaseModel

from app.services.pricing_service import DEFAULT_PRINTING_MODE, DEFAULT_PRINTING_SIDES


class PricingCalculateRequest(BaseModel):
    page_count: int
    printing_mode: str = DEFAULT_PRINTING_MODE
    printing_sides: str = DEFAULT_PRINTING_SIDES
    acknowledgment_of_receipt: bool = False


class PricingBreakdownRead(BaseModel):
    page_count: int
    sheet_count: int
    printing_mode: str
    printing_sides: str
    paper_weight_g: str
    envelope_weight_g: str
    estimated_weight_g: str
    weight_bracket: str
    printing_cost: str
    paper_cost: str
    envelope_cost: str
    postal_postage: str
    registered_mail_fee: str
    acknowledgment_of_receipt: bool
    acknowledgment_fee: str
    delivery_fee: str
    service_fee: str
    total: str
    currency: str

File: frontend/src/types/pricing.ts
export interface PricingBreakdown {
  page_count: number
  sheet_count: number
  printing_mode: string
  printing_sides: string
  paper_weight_g: string
  envelope_weight_g: string
  estimated_weight_g: string
  weight_bracket: string
  printing_cost: string
  paper_cost: string
  envelope_cost: string
  postal_postage: string
  registered_mail_fee: string
  acknowledgment_of_receipt: boolean
  acknowledgment_fee: string
  delivery_fee: string
  service_fee: string
  total: string
  currency: string
}
